validator/sawtooth_validator/state/state_delta_processor.py

```diff
@@ -188,19 +236,41 @@ def handle(self, connection_id, message_content):
         request.ParseFromString(message_content)
 
         ack = RegisterStateDeltaSubscriberResponse()
+        if self._delta_processor.is_valid_subscription(
+                request.last_known_block_ids):
+            ack.status = ack.OK
+            return HandlerResult(
+                HandlerStatus.RETURN_AND_PASS,
+                message_out=ack,
+                message_type=self._msg_type)
+        else:
+            ack.status = ack.UNKNOWN_BLOCK
+            return HandlerResult(
+                HandlerStatus.DROP,
```

Comment:
HandlerStatus.DROP will stop processing on the message, and not send a message back, but since an ack is being sent back a HandlerStatus.RETURN is needed here. RETURN stops with that handler for that message, but sends a message back.